Common/src/geometry/meshreader/CSU2ASCIIMeshReaderFVM.cpp

```diff
@@ -718,705 +740,304 @@ void CSU2ASCIIMeshReaderFVM::SplitActuatorDiskSurface() {
+      bool isOwned = false;
+      for (unsigned short i = 0; i < nPointsElem; i++) {
+        if (pointPartitioner.IndexBelongsToRank(connectivity[i], rank)) {
+          isOwned = true;
+          break;
+        }
+      }
 
-            /* If so, we need to store the element locally. */
-            if (isOwned) {
-              localVolumeElementConnectivity.push_back(GlobalIndex);
-              localVolumeElementConnectivity.push_back(VTK_Type);
-              for (unsigned short i = 0; i < N_POINTS_HEXAHEDRON; i++) {
-                localVolumeElementConnectivity.push_back(connectivity[i]);
-              }
-              numberOfLocalElements++;
-            }
-            GlobalIndex++;
-            break;
+      /* If element is owned, we need to store it locally. */
+      if (isOwned) {
+        localVolumeElementConnectivity.push_back(GlobalIndex);
+        localVolumeElementConnectivity.push_back(VTK_Type);
+        /// TODO: Use a compressed format.
+        for (unsigned short i = 0; i < N_POINTS_HEXAHEDRON; i++) {
+          localVolumeElementConnectivity.push_back(connectivity[i]);
         }
+        numberOfLocalElements++;
       }
-      if (GlobalIndex == numberOfGlobalElements) break;
     }
+    break;
   }
-
-  mesh_file.close();
-
 }
 
-void CSU2ASCIIMeshReaderFVM::ReadSurfaceElementConnectivity() {
+void CSU2ASCIIMeshReaderFVM::ReadSurfaceElementConnectivity(const bool single_pass) {
 
   /* We already read in the number of markers with the metadata. */
   surfaceElementConnectivity.resize(numberOfMarkers);
   markerNames.resize(numberOfMarkers);
 
-  vector<unsigned long> connectivity(N_POINTS_HEXAHEDRON,0);
+  array<unsigned long, N_POINTS_HEXAHEDRON> connectivity{};
 
   /*--- In this routine, the boundary info is read by all ranks,
    however, the surface connectivity is still handled by the
    master node (and eventually distributed by the master as well). ---*/
 
-  mesh_file.open(meshFilename, ios::in);
-  FastForwardToMyZone();
-
-  string text_line;
-  string::size_type position;
-  bool foundMarkers = false;
-  while (getline (mesh_file, text_line)) {
-
-    /*--- Jump to the section containing the markers. ---*/
-
-    position = text_line.find ("NMARK=",0);
-    if (position != string::npos) {
-
-      foundMarkers = true;
-      unsigned short iMarker = 0;
-      while (iMarker < numberOfMarkers) {
-        getline (mesh_file, text_line);
-        text_line.erase (0,11);
-        string::size_type position;
-
-        for (unsigned short iChar = 0; iChar < 20; iChar++) {
-          position = text_line.find( " ", 0 );
-          if (position != string::npos) text_line.erase (position,1);
-          position = text_line.find( "\r", 0 );
-          if (position != string::npos) text_line.erase (position,1);
-          position = text_line.find( "\n", 0 );
-          if (position != string::npos) text_line.erase (position,1);
-        }
```

Comment:
Comes from the ancient ones, I suppose we strip up to 20 spaces anywhere from the marker names :shrug: 
I doubt the \r and \n ever do anything, because these are used by getline to know what is a line.
This should not be performance or correctness critical, so if it ain't broke...